Overview of "*Extra_Analysis*" folder contents

 - [Age_Analysis](https://github.com/cwarden45/HPV_genotype_paper-archived_samples/tree/master/Downstream_R_Code/Extra_Analysis/Age_Analysis) - a variety of statistical analysis to check for differences based upon patient age
 - [Ancestry_Analysis](https://github.com/cwarden45/HPV_genotype_paper-archived_samples/tree/master/Downstream_R_Code/Extra_Analysis/Ancestry_Analysis) - a variety of statistical analysis to check for differences based upon patient estimated ancestry
 - [Collection_Date_Analysis](https://github.com/cwarden45/HPV_genotype_paper-archived_samples/tree/master/Downstream_R_Code/Extra_Analysis/Collection_Date_Analysis) - a variety of statistical analysis to check for differences based upon collection date of the patient's sample
 - [Confirming-Completing_Patient_Annotations](https://github.com/cwarden45/HPV_genotype_paper-archived_samples/tree/master/Downstream_R_Code/Extra_Analysis/Confirming-Completing_Patient_Annotations) - justification for using plink Identity-By-Descent (IBD) for identifying samples that come from the same patient
 - [Cross-Contamination_Into_Samples_From_Other_Labs](https://github.com/cwarden45/HPV_genotype_paper-archived_samples/tree/master/Downstream_R_Code/Extra_Analysis/Cross-Contamination_Into_Samples_From_Other_Labs) - recovery of exact matches to the PEAR-merged L1 Amplicon in other lab's samples (where that exact sequence should be rare, or likely non-existent if there was not an index hopping)
 - [Effect_of_Human_Read_Threshold_on_Genotypes](https://github.com/cwarden45/HPV_genotype_paper-archived_samples/tree/master/Downstream_R_Code/Extra_Analysis/Effect_of_Human_Read_Threshold_on_Genotypes) - comparision of 4 sets of thresholds relative to off-target human reads to see the effect of HPV16/HPV18/HPV58 assignments (and "unclear" genotype assignments.  Repeated for the following other conditions
   - minimum 5% read fractions (overall + genotype), without qPCR filter
   - minimum 20% read fractions (overall + genotype), without qPCR filter
   - minimum 20% read fractions (overall + genotype), **with** qPCR filter
 - [Effect_of_qPCR_Flag_on_Co-Infections](https://github.com/cwarden45/HPV_genotype_paper-archived_samples/tree/master/Downstream_R_Code/Extra_Analysis/Effect_of_qPCR_Flag_on_Co-Infections) - effect of using 5% read fraction, 20% read fraction, and 50% read fraction thresholds on co-infections.  Co-infections decrease with more stringent thresholds.  Repeated for the following conditions:
   - without qPCR filter
   - **with** qPCR filter
 - [Effect_of_qPCR_Flag_on_HPV58_Genotypes](https://github.com/cwarden45/HPV_genotype_paper-archived_samples/tree/master/Downstream_R_Code/Extra_Analysis/Effect_of_qPCR_Flag_on_HPV58_Genotypes) - effect of using 5% read fraction, 20% read fraction, plurality and majority (50% read fraction) thresholds on HPV58 genotype assignments.  Variability in HPV58 assignments between sample types decreases with more stringent thresholds.  Repeated for the following conditions:
   - without qPCR filter
   - **with** qPCR filter
 - [Effect_of_qPCR_Flag_on_Tumor-Tumor_Divergence](https://github.com/cwarden45/HPV_genotype_paper-archived_samples/tree/master/Downstream_R_Code/Extra_Analysis/Effect_of_qPCR_Flag_on_Tumor-Tumor_Divergence) - plots of paired sample correlations without and with qPCR filter.  HPV16 correlation improves with qPCR filter, and less reproducible HPV58+ pairs are removed with qPCR filter
 - [Human-Read_Adjusted_Divergence](https://github.com/cwarden45/HPV_genotype_paper-archived_samples/tree/master/Downstream_R_Code/Extra_Analysis/Human-Read_Adjusted_Divergence) - a visual representation of the effect of the human-read thresolds (somewhat similar to [Effect_of_Human_Read_Threshold_on_Genotypes](https://github.com/cwarden45/HPV_genotype_paper-archived_samples/tree/master/Downstream_R_Code/Extra_Analysis/Effect_of_Human_Read_Threshold_on_Genotypes), but focusing on correlation coefficients)
 - [Low_Frequency_Sample_Size_Calculation](https://github.com/cwarden45/HPV_genotype_paper-archived_samples/tree/master/Downstream_R_Code/Extra_Analysis/Low_Frequency_Sample_Size_Calculation) - simplified sample size calculation for rare genotype differences (15% versus 5% population frequency, and 10% versus 5% population frequency)
 - [Median_Off-Target_Human_Insert_Size_Distribution](https://github.com/cwarden45/HPV_genotype_paper-archived_samples/tree/master/Downstream_R_Code/Extra_Analysis/Median_Off-Target_Human_Insert_Size_Distribution) - distribution of insert size calculated from off-target human reads; matches what is expected from gel extraction, but there were still statistically significant differences between the sample types.
 - [Representative_Sequence_Analysis](https://github.com/cwarden45/HPV_genotype_paper-archived_samples/tree/master/Downstream_R_Code/Extra_Analysis/Representative_Sequence_Analysis) - attempts to understand mechanisms related to reproducilbity of read fractions for HPV16/HPV18/HPV58.  Two different type of strategies are used:
   - Number of human genome BLAT hits (from HPV L1 Amplicon sequences), with annotations of overlaping repeat elements
   - Exact sequence fractions for specific HPV genotypes; SNPs found in FFPE samples for HPV16 and HPV18 (but not HPV58)
 - [Sample_Type_Full_Statistical_Analysis](https://github.com/cwarden45/HPV_genotype_paper-archived_samples/tree/master/Downstream_R_Code/Extra_Analysis/Sample_Type_Full_Statistical_Analysis) - a variety of statistical analysis to check for differences based upon the archive type of the sample
 - [qPCR-Filtered_HPV-Genotype_Human-Read_Correlation](https://github.com/cwarden45/HPV_genotype_paper-archived_samples/tree/master/Downstream_R_Code/Extra_Analysis/qPCR-Filtered_HPV-Genotype_Human-Read_Correlation) - visualization of correlation for off-target human reads versus HPV genotype read fractions (before and after qPCR filter).  Trendlines similar, but you can see qPCR filtered FFPE samples in the HPV58 plot.
