De-Multiplexing Considerations for Our L1 Amplicon-Seq Samples
-----------------
Although only a small fraction of samples were affected, we observed that both technical and human error influenced the accuracy of our results prior to our in-depth analysis. For example, we observed some evidence of cross contamination in samples from other laboratories that were processed with ours (Additional File 7: Figure S2). This is likely the result of “index hopping” [43-45], but we also withdrew one sample from our Sequence Read Archive (SRA) deposit because a sample from another laboratory accidentally had the barcode intended for an HPV L1 amplicon sequencing sample, which was due to human rather than technical error. Hypothetically, cross-contamination upstream of library preparation could also be called human error, but that could not be improved through base calling or sequencing conditions (although use of more conservative genotyping thresholds may help in that situation). However, this section focuses on how the sequencing step could affect HPV58 co-infections.

Cross-Contamination of Our L1 Amplicon-Seq Sequences into Other Lab's Samples
-----------------

![HPV L1 Amplicon Exact Matches](2017-FFPE-Runs_HPV-L1-Amplicon-Cross-Contamination.png "HPV L1 Amplicon Exact Matches")

The archived DNA and frozen tissues were from 2016 runs.  Since we could not recover those run folders, we could not do a similar comparison for those runs.

Nevertheless, the exact HPV L1 amplicon sequence also provides a measure of cross-contamination into other samples, which we can see in samples for other laboratories that shared a lane with samples from this project. We observed that one sample from another laboratory, which was not expected to have any L1 amplicon sequencing reads but shared a lane with samples from this project, had high HPV58 percentages (~10%) in each of the 2 lanes used for sequencing. This sample was processed in a run with 26 FFPE L1 amplicon sequencing samples (Run 271, with barcode GACCGG).  We believe this sample also had the wrong barcode (or otherwise had a QC issue, with few fragments to be sequenced), so it was incorrectly de-multiplexed.  This resulted in substantially fewer reads (<2000 total) compared to the L1 amplicon sequencing samples (311,370–3,757,409) and most other samples from the other laboratory (5–10 million).

This may be important if samples with fewer starting reads (such as the HPV L1 amplicon sequencing samples) may greater cross-contamination from de-multiplexing issues than most other samples processed on our HiSeq2500 machines (unless de-multiplexing is skipped or reduced altogether). While ~10% HPV58 sample from another lab was not supposed to have any L1 Amplicon-Sequencing reads (and therefore would have been called HPV-, if that was a human sample), this is important because this is the read fraction range (5–50%) in which we see HPV58 co-infections at a higher rate than previously reported. So, this specifically shows that we have observed HPV58 co-infections as false positives in the exact same run as some of our FFPE L1 Amplicon Sequencing samples (with the increased HPV58 co-infection rate), but typical index hopping likely has a greater effect on HPV58 co-infections with read fractions less than 15%. We would usually expect it to be lower (less than 5%) [44], although perhaps including samples from other labs didn’t entirely make up for having a low percent PhiX spike-in [46] (<2%) for a low-complexity library and/or there are special complications when sequencing HPV L1 Amplicons.
